ipatests/test_integration/test_backup_and_restore.py

```diff
@@ -434,3 +449,107 @@ class TestBackupReinstallRestoreWithKRA(BaseBackupAndRestoreWithKRA):
     def test_full_backup_reinstall_restore_with_vault(self):
         """backup, uninstall, reinstall, restore"""
         self._full_backup_restore_with_vault(reinstall=True)
+
+
+class TestReplicaInstallAfterRestore(IntegrationTest):
+    """Test to check second replica installation after master restore
+
+    When master is restored from backup and replica1 is re-initialize,
+    second replica installation was failing. The issue was with ipa-backup
+    tool which was not backing up the /etc/ipa/custodia/custodia.conf and
+    /etc/ipa/custodia/server.keys.
+
+    related ticket: https://pagure.io/freeipa/issue/7247
+    """
+
+    num_replicas = 2
+
+    def test_replica_install_after_restore(self):
+        master = self.master
+        replica1 = self.replicas[0]
+        replica2 = self.replicas[1]
+        test1 = "test1"
+        test2 = "test2"
+
+        tasks.install_master(master)
+        tasks.install_replica(master, replica1)
+
+        result = user_add(master, test1)
+        assert result.returncode == 0
+
+        result = user_add(replica1, test2)
+        assert result.returncode == 0
+
+        tasks.wait_for_replication(replica1.ldap_connect())
+        result = user_show(replica1, test1)
+        assert result.returncode == 0
+
+        result = user_show(master, test2)
+        assert result.returncode == 0
+
+        # backup master
+        backup_path = backup(master)
+
+        domain = replica1.domain.name.split('.')
+        tf = NamedTemporaryFile()
+        ldif_file = tf.name
+        entry_ldif = textwrap.dedent(
+            "dn: cn=meTo{hostname},cn=replica,"
+            "cn=dc\\3D{sub_domain}\\2Cdc\\3D{domain},"
```

Comment:
Hi @mrizwan93, 
this template will not work for all the cases (for instance if domain contains more than 2 levels like ipadomain.example.com). You can probably use ipautil.realm_to_suffix and will need to escape the special chars (ldap.dn provides escape_dn_chars method).